### ISSUE TITLE
MFA Authentication doesn't work

### ISSUE THREAD

**Issue by toddysm** [Microsoft/PartnerCenterPowerShellModule#37]
Trying to use the PS Module to fetch information about Partner Center users fails:

```
PS C:\Users\toddy> $credentials = Get-Credential 'toddysm@agitaretech.com'
PS C:\Users\toddy> Get-PCCustomer -all
>>>Use Add-PCAuthentication function to login to partnercenter or use -satoken parameter<<<
At C:\Program Files\WindowsPowerShell\Modules\PartnerCenterModule\0.9.0.14\commons.ps1:45 char:9
+         throw ">>>Use Add-PCAuthentication function to login to partn ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (>>>Use Add-PCAu...en parameter<<<:String) [], RuntimeException
    + FullyQualifiedErrorId : >>>Use Add-PCAuthentication function to login to partnercenter or use -satoken parameter<<<
```

My assumption is that this module doesn't support MFA and my account has MFA enforced or the documentation is not clear how to use the module. The `-satoken` parameter error is quite confusing.

Any suggestions what to do to resolve those.

**Comment by toddysm** [comment]
Changed the title to be more clear what the issue is. Here is the error when authenticating:
```
Invoke-RestMethod : {"error":"interaction_required","error_description":"AADSTS50076: Due to a configuration change made by your administrator, or
because you moved to a new location, you must use multi-factor authentication to access 'SampleBECApp'.\r\nTrace ID:
82eb9a17-9069-4708-9f59-033e802e0800\r\nCorrelation ID: da06dd53-c08f-4a66-b684-f2deb9a751cf\r\nTimestamp: 2017-12-20
22:56:20Z","error_codes":[50076],"timestamp":"2017-12-20
22:56:20Z","trace_id":"82eb9a17-9069-4708-9f59-033e802e0800","correlation_id":"da06dd53-c08f-4a66-b684-f2deb9a751cf"}
At C:\Program Files\WindowsPowerShell\Modules\PartnerCenterModule\0.9.0.14\PartnerCenterAuthentication.psm1:32 char:17
+ ... $response = Invoke-RestMethod -Uri $url -ContentType "application/x-w ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
The property 'access_token' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PartnerCenterModule\0.9.0.14\PartnerCenterAuthentication.psm1:33 char:12
+     return $response.access_token
+            ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

The property 'Domain' cannot be found on this object. Verify that the property exists.
At C:\Program Files\WindowsPowerShell\Modules\PartnerCenterModule\0.9.0.14\PartnerCenterAuthentication.psm1:104 char:5
+     Update-ModuleTelemetry -cspDomain $result.Domain
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict

Invoke-RestMethod : {"Message":"An error has occurred."}
At C:\Program Files\WindowsPowerShell\Modules\PartnerCenterModule\0.9.0.14\PartnerCenterTelemetry.psm1:31 char:19
+ ...   $result = Invoke-RestMethod -Method Post -Uri "https://api.migaz.to ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

**Comment by toddysm** [comment]
This can be worked around as suggested in #21 using AppID and App Secret, however, there was a bug that should be fixed once PR #38 is integrated
